### PR TITLE
Instagram integration

### PR DESCRIPTION
**Pull request by zeus999**
* instagram callback component
* instagram integration service

### CODE REVIEW

**Comment by ashish173** [comment]
@zeus999 some merge conflicts pls resolve

File: src/app/shared/header/header.component.ts [hunk 1]
```diff
@@ -18,6 +18,10 @@ import "rxjs/add/operator/let";
 export class HeaderComponent implements OnInit {
   user$: Observable<UserProfile>;
   authentication$: Observable<any>;
+  CLIENT_ID: string = "865ffa528b874fc3b755ee13b9a79037";
```
Comment (ashish173) [inline src/app/shared/header/header.component.ts]: we are adding secret in the file pls make this change,

File: src/app/instagram-integration.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  exchangeCodeWithToken(code: string): Observable<string> {
+    const headers = new Headers({
+      'Content-Type': 'application/json',
+      'Authorization': this.auth_token
+    })
+    return this.http.post(`${this.apiLink}/exchange_code_with_token`, {code: code}, { headers: headers })
+		  .map(data => data.json());
+  }
+
+  getUserInstagramMedia(): Observable<{}> {
```
Comment (ashish173) [inline src/app/instagram-integration.service.ts]: maybe it should be `getUserInstagramMedia(): Observable<Object> {` @pkrawat1 @kartikjagdale what do you think?

File: src/app/components/trips/trip-detail/trip-activity/trip-activity.component.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { LoadMediaAction } from './../../../../actions/instagram.action';
 import { City } from './../../../../models/city';
```
Comment (ashish173) [inline src/app/components/trips/trip-detail/trip-activity/trip-activity.component.ts]: how come HTML in code where there should be imports

File: src/app/components/instagram-authentication-callback/instagram-authentication-callback.component.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+export class InstagramAuthenticationCallbackComponent implements OnInit {
+
+  exchangeTokenStatus: string;
+  codeStatus: boolean = false;
+
+  constructor(private router: Router, private insta: InstagramIntegrationService) { }
+
+  ngOnInit() {
+    const queryParams = this.router.routerState.snapshot.root.queryParams;
+		const code = queryParams['code'];
```
Comment (ashish173) [inline src/app/components/instagram-authentication-callback/instagram-authentication-callback.component.ts]: indentation here

File: src/app/components/instagram-authentication-callback/instagram-authentication-callback.component.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    const error = queryParams['error'];
+    const error_description = queryParams['error_description'];
+    
+    if(code){
+      this.codeStatus = true;
+			this.insta.exchangeCodeWithToken(code)
```
Comment (ashish173) [inline src/app/components/instagram-authentication-callback/instagram-authentication-callback.component.ts]: indentation here too

**Comment by ashish173** [comment]
@zeus999 In general, I could see a lot of HTML code for imports on top of the files could you pls investigate 😁

File: src/app/components/instagram-authentication-callback/instagram-authentication-callback.component.ts (hunk 5 above)
Comment (pkrawat1) [inline src/app/components/instagram-authentication-callback/instagram-authentication-callback.component.ts]: rt click and Document format before saving

File: src/app/instagram-integration.service.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+// This method will exchange code with instagram access token
+// for more information, visit https://www.instagram.com/developer/authentication/
+  exchangeCodeWithToken(code: string): Observable<string> {
+    const headers = new Headers({
+      'Content-Type': 'application/json',
+      'Authorization': this.auth_token
+    })
+    return this.http.post(`${this.apiLink}/exchange_code_with_token`, {code: code}, { headers: headers })
```
Comment (pkrawat1) [inline src/app/instagram-integration.service.ts]: https://github.com/2muchcoffeecom/ng2-restangular

**Pull request merged**